resources/views/partials/modules/header.blade.php

```diff
@@ -1,24 +1,40 @@
 <header class="py-8 bg-gray-700">
-    <div class="max-w-screen-xl px-4 mx-auto sm:px-6 lg:px-8">
+    <div class="flex flex-col max-w-screen-xl px-4 mx-auto sm:items-center sm:justify-between sm:flex-row-reverse sm:px-6 lg:px-8">
+
         @if (Config::has('localized-routes.supported-locales'))
-            <select onchange="location = this.value;" class="float-right max-w-6xl px-4 mx-auto text-xs text-black bg-gray-200 border border-gray-200 rounded sm:px-6 lg:px-8 lang">
-                @foreach (Config::get('localized-routes.supported-locales', []) as $lang)
-                    <option value="{{ Route::localizedUrl($lang) }}" @if (App::getLocale() == $lang) selected="selected" @endif class="lang-name lang-name-{{ $lang }}">
-                        {{ Config::get("localized-routes.locales-name-native.{$lang}", Str::upper($lang)) }}
-                    </option>
-                @endforeach
-            </select>
+            @php
+                $formatted_languages = [];
+                $current_language = null;
+                
+                foreach (Config::get('localized-routes.supported-locales', []) as $lang) {
+                    $language_name_native = Config::get("localized-routes.locales-name-native.{$lang}", Str::upper($lang));
+                    $formatted_languages[] = [
+                        'language_code' => $lang,
+                        'language_name_native' => $language_name_native,
+                        'language_url' => Route::localizedUrl($lang),
+                    ];
+                    if (App::getLocale() == $lang) {
+                        $current_language = $language_name_native;
+                    }
+                }            
+            @endphp
```

Comment:
I know this will sound crazy, but what if we move this entire block (including the vue component) into a Laravel component, and then you could move all this PHP into the Class for the Laravel component?

Reply:
@mattstauffer Completed!